src/decorators/usePreventLeaveDecorator/usePreventLeaveDecorator.ts

```diff
@@ -55,11 +55,19 @@ export const usePreventLeaveDecorator = (
 
   return React.useCallback<Decorator<any, any>>((form) => {
     const subscriber: Subscriber<FormState<any>> = (state) => {
-      dirtySinceLastSubmitRef.current = state.dirtySinceLastSubmit
+      /*
+       * dirtySinceLastSubmit is false until we submit the form when pristine keeps initial initialValues
+       * for references
+       */
+      shouldPreventLeaving.current = state.submitSucceeded
+        ? state.dirtySinceLastSubmit
+        : !state.pristine
```

Comment:
I would have used `dirty` here.
If my memory serves me well, I think there's another hook that might benefit from this refinement.

Reply:
If I understand well the documentation, `dirty === !pristine` but I can change it :)